app/server/utils/transfer_limits.py

```diff
@@ -0,0 +1,251 @@
+from typing import List, Callable, Optional, Union, Tuple
+import datetime
+from toolz import curry, pipe
+from sqlalchemy import or_
+from sqlalchemy.orm import Query
+
+from server.models import token
+from server.models.credit_transfer import CreditTransfer
+from server.utils.transfer_enums import TransferSubTypeEnum, TransferTypeEnum
+
+AppliedToTypes = List[Union[TransferTypeEnum,
+                            Tuple[TransferTypeEnum, TransferSubTypeEnum]]]
+
+PAYMENT = TransferTypeEnum.PAYMENT
+DEPOSIT = TransferTypeEnum.DEPOSIT
+WITHDRAWAL = TransferTypeEnum.WITHDRAWAL
+
+AGENT_OUT = TransferSubTypeEnum.AGENT_OUT
+STANDARD = TransferSubTypeEnum.STANDARD
+
+STANDARD_PAYMENT = (PAYMENT, STANDARD)
+AGENT_OUT_PAYMENT = (PAYMENT, AGENT_OUT)
+
+
+def get_transfer_limits(credit_transfer: CreditTransfer):
+    relevant_limits = []
+    for limit in LIMITS:
+        applied = limit.application_filter(credit_transfer)
+        if applied and (credit_transfer.transfer_type in limit.applied_to_transfer_types
+                        or (credit_transfer.transfer_type, credit_transfer.transfer_subtype)
+                        in limit.applied_to_transfer_types):
+            
+            relevant_limits.append(limit)
+
+    return relevant_limits
+
+
+    # # Supports filtering over type-subtype tuples of the form ('PAYMENT', 'AGENT_OUT')
+    # applied = limit.application_filter(self)
+    # if applied:
+    #     for transfer_type in limit.applied_to_transfer_types:
+    #         if isinstance(transfer_type, (tuple, list)):
+    #             if str(self.transfer_type) == transfer_type[0]\
+    #                     and str(self.transfer_subtype) == transfer_type[1]:
+    #                 relevant_limits.append(limit)
+    #                 continue
+    #         else:
+    #             if str(self.transfer_type) == transfer_type:
+    #                 relevant_limits.append(limit)
+    #                 continue
```

Comment:
?

Reply:
ohh just an alternate implementation that i didn't wanna throw out just yet.